### ISSUE TITLE
Cannot find my extension widget js after upgrading to 2.4.0

### ISSUE THREAD

**Issue by maeglin89273** [bokeh/bokeh#11619]
I built my extension widgets following the example in the bokeh repository (custom/wrapped_extension_full).
It worked fine in 2.3.3. The browser could fetch my js extension file successfully under the URL /static/extensions/ .
After I upgraded to 2.4.0, I issued the bokeh command to rebuild it. The js extension file was built successfully in the usual dist folder, but the browser no longer could fetch it.

Any changes I missed about the extension path in 2.4.0?

**Comment by bryevdv** [comment]
cc @mattpap 

@maeglin89273 we really have to have a complete [Minimal Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) in order to actually investigate.

**Comment by maeglin89273** [comment]
Yes @bryevdv, I just created an minimum project to reproduce the issue as the following attachment. I just realized I didn't mentioned the extension was running in **server form**.
Basically, I adopted the custom/wrapped_extension_base example into the bokeh server form. This example can run in 2.3.3 but not 2.4.0.
I built the extension by running ```bokeh build``` in the latex_label directory, and run ```latex_extension_base.py``` to launch the server.
For 2.3.3 environment compatibility reasons, I removed ```override``` keywords in class ```LatexLabel``` and revert the static block to static ```init_LatexLabel``` method.

[bokeh-issue-11619-example-cleaned.zip](https://github.com/bokeh/bokeh/files/7175198/bokeh-issue-11619-example-cleaned.zip)

**Comment by bryevdv** [comment]
@mattpap any thoughts about any relevant changes that could explain this?

**Comment by maeglin89273** [comment]
I just found the root cause. 
The extension url pattern `/static/extensions/(.*)` is declared in [bokeh/server/urls.py](https://github.com/bokeh/bokeh/blob/4f121520d586459ec835d4d676ee4fd34ccc4cc5/src/bokeh/server/urls.py#L112) module, which is added after another static handler route located in [bokeh/server/tornado.py](https://github.com/bokeh/bokeh/blob/4f121520d586459ec835d4d676ee4fd34ccc4cc5/src/bokeh/server/tornado.py#L435). This changes the behavior before the version 2.4.0 that conditionally adds this route. 
The key commit is [f92f42](https://github.com/bokeh/bokeh/commit/f92f42f275d61069954ade63e56f2570ad6ce010). Additionally, there already exists a static handler route in [bokeh/server/urls.py](https://github.com/bokeh/bokeh/blob/4f121520d586459ec835d4d676ee4fd34ccc4cc5/src/bokeh/server/urls.py#L112), evaluated after the extension handler.

 The handler precedence behind the Tornado server is that it picks the first matched handler from the list(`all_patterns`), and therefore the extension url pattern never get evaluated. Here is the my stack trace evidence:
![Screenshot from 2023-01-09 12-49-56](https://user-images.githubusercontent.com/5237288/211242879-eefd0903-5f48-4815-9b0f-f45f778a8ef2.png)

**Comment by bryevdv** [comment]
@maeglin89273 I should advise at this point that there will almost certainly not be any more 2.x releases at all. Can you confirm that this issue persists in Bokeh 3.0.3? If so I will re-triage this as a `bug` and target 3.x milestone. Since you have already dived into this, are you interested to submit a PR for consideration?

**Comment by maeglin89273** [comment]
Yes, actually this walkthrough was done on version 3.0.3. Sorry I didn't mention that, and I'm not confident to submit a PR.
